src/dry-run-q.ts

```diff
@@ -0,0 +1,148 @@
+} from './constants.js';
+import { spawn } from 'child_process';
+import { getOctokit } from './octokit.js';
+
+let jobQueue = Promise.resolve();
+
+const SHERIFF_DRY_RUN_CHECK_NAME = 'Sheriff Dry Run';
+
+export async function queueDryRun(octokit: Octokit, mergeSha: string | null, headSha: string) {
+  if (!mergeSha) {
+    await octokit.checks.create({
+      repo: PERMISSIONS_FILE_REPO,
+      owner: PERMISSIONS_FILE_ORG,
+      name: SHERIFF_DRY_RUN_CHECK_NAME,
+      head_sha: headSha,
+      status: 'completed',
+      conclusion: 'failure',
+      completed_at: new Date().toISOString(),
+      output: {
+        title: 'Dry Run Not Possible',
+        summary: 'No merge sha available',
+      },
+    });
+    return;
+  }
+  await octokit.checks.create({
+    repo: PERMISSIONS_FILE_REPO,
+    owner: PERMISSIONS_FILE_ORG,
+    name: SHERIFF_DRY_RUN_CHECK_NAME,
+    head_sha: headSha,
+    status: 'in_progress',
+    started_at: new Date().toISOString(),
+  });
+
+  jobQueue = jobQueue
+    .then(async () => {
+      const configPath = path.resolve(os.tmpdir(), `sheriff-${mergeSha}-${headSha}.yaml`);
+      const fileContent = await octokit.repos.getContent({
+        repo: PERMISSIONS_FILE_REPO,
+        owner: PERMISSIONS_FILE_ORG,
+        ref: mergeSha,
+        path: PERMISSIONS_FILE_PATH,
+      });
+      if (Array.isArray(fileContent.data)) {
+        // Nah you bad
+        throw new Error('wat');
+      }
+
+      await fs.promises.writeFile(
+        configPath,
+        // @ts-ignore - Octokit fails to type properties of ReposGetContentsResponse correctly.
+        Buffer.from(fileContent.data.content || '', fileContent.data.encoding as any).toString(
+          'utf8',
+        ),
+      );
+
+      let out = '';
+      const success = await new Promise<boolean>((resolve) => {
+        const child = spawn(
+          process.execPath,
+          [path.resolve(import.meta.dirname, 'permissions', 'run.js')],
+          {
+            stdio: 'pipe',
+            env: {
+              ...process.env,
+              PERMISSIONS_FILE_LOCAL_PATH: configPath,
+              FORCE_COLOR: '2',
+            },
+          },
+        );
+
+        child.stdout.on('data', (d) => {
+          out += d.toString();
+        });
+        child.stderr.on('data', (d) => {
+          out += d.toString();
+        });
+
+        child.on('exit', (code, signal) => {
+          if (signal !== null) return resolve(false); // Unknown error
+          if (code === 0) return resolve(true); // We good
+          resolve(false);
+        });
+      });
+
+      const svg = ansiToSvg(out, {
+        paddingTop: 16,
+        paddingLeft: 16,
+        paddingRight: 16,
+        paddingBottom: 16,
+      });
+      const gistOctokit = new Octokit({
+        auth: SHERIFF_GIST_TOKEN,
+      });
+      const gistData = await gistOctokit.gists.create({
+        files: {
+          ['out.svg']: {
+            content: svg,
+          },
+        },
+      });
```

Comment:
Where does this Gist live when a GH app is running this code? The API documentation only lists "GitHub App user access tokens", not installation tokens.

Reply:
I guess that's what `SHERIFF_GIST_TOKEN` is for? Can that be documented?

Reply:
> I guess that's what SHERIFF_GIST_TOKEN is for? 

yeah, I don't love it, but wanted to get this working before wasting too many cycles figuring out a better way to store SVG files 🙃